File: backend/Titan.API/Services/UserService.cs
using Microsoft.EntityFrameworkCore;
using Titan.API.Data;
using Titan.API.DTOs;

namespace Titan.API.Services;

public class UserService : IUserService
{
    private readonly TitanDbContext _context;

    public UserService(TitanDbContext context)
    {
        _context = context;
    }
    public async Task<bool> UpdateUsernameAsync(int userId, string newUserName)
    {
        var user = await _context.Users.FirstOrDefaultAsync(u => u.Id == userId);

        if (user == null) return false;

        user.Username = newUserName;

        await _context.SaveChangesAsync();

        return true;
    }

    public async Task<IEnumerable<UserDto>> GetAllEmployeesAsync()
    {
        return await _context.Users
        .Where(u => u.Role == "Employee")
        .Select(u => new UserDto
        {
            Id = u.Id,
            Email = u.Email,
            UserName = u.Username,
            Role = u.Role
        })
        .ToListAsync();
    }

}

File: frontend/src/services/userService.js
import api from "./api";

const userService = {

    getAllEmployees: async () => {
        const response = await api.get('/users/employees');

        return response.data;
    },

    updateMyUsername: async (userName) => {
        const response = await api.put('/users/me/username', {
            userName
        })
        return response.data;
    }
}

export default userService;